src/mono/mono/metadata/appdomain.c

```diff
@@ -725,7 +725,8 @@ mono_domain_assembly_preload (MonoAssemblyLoadContext *alc,
 	req.request.predicate = predicate;
 	req.request.predicate_ud = predicate_ud;
 
-	if (!mono_runtime_get_no_exec ()) {
+        // only look in ApplicationBase for satellite assemblies
+	if (!mono_runtime_get_no_exec () && !mono_assembly_name_culture_is_neutral (aname)) {
```

Comment:
Why the satellite assembly special case?   As far as I can tell, CoreCLR doesn't use ApplicationBase for those either ...

Reply:
Yea - that was wrong. I was guessing based on the test failures.